actionpack/lib/abstract_controller/rendering.rb

```diff
@@ -64,12 +64,12 @@ def rendered_format
     # You can overwrite this configuration per controller.
     def view_assigns
       protected_vars = _protected_ivars
-      variables      = instance_variables
 
-      variables.reject! { |s| protected_vars.include? s }
-      variables.each_with_object({}) { |name, hash|
-        hash[name.slice(1, name.length)] = instance_variable_get(name)
-      }
+      instance_variables.each_with_object({}) do |name, hash|
+        unless protected_vars.include?(name)
+          hash[name[1..-1]] = instance_variable_get(name)
```

Comment:
Why changed `name.slice(1, name.length)` to `name[1..-1]`? `1..-1` allocates an extra range object so seems slow than before.

Reply:
Is that the case? I benchmarked memory as well and extra allocations didn't show up. Maybe I'm missing something?

### Memory benchmark
```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  gem "rails", github: "rails/rails", require: "rails/all"
  gem "benchmark-ips"
  gem "benchmark-memory"
end

module AbstractController
  module Rendering
    def fast_view_assigns
      protected_vars = _protected_ivars

      instance_variables.each_with_object({}) do |name, hash|
        unless protected_vars.include?(name)
          hash[name[1..-1]] = instance_variable_get(name)
        end
      end
    end
  end
end

class ApplicationController < ActionController::Base
end

controller = ApplicationController.new
controller.perform_caching = false
controller.locale = :en
controller.logger = Logger.new(STDOUT)
controller.action_name = :index
controller.params = { a: 1, b: 2 }
controller.etag_with_template_digest = "some_etag"
controller._wrapper_options = { a: 1 }
controller.action_has_layout = true
controller._helper_methods = %i[:validate_something]
controller._view_cache_dependencies = 1
controller._renderers = ApplicationController.renderer
controller.include_all_helpers = true
controller.default_url_options = { param: "value" }

Benchmark.memory do |x|
  x.report("view_assigns")      { controller.view_assigns }
  x.report("fast_view_assigns") { controller.fast_view_assigns }
  x.compare!
end
```
### Results
```
Calculating -------------------------------------
        view_assigns   688.000  memsize (     0.000  retained)
                        10.000  objects (     0.000  retained)
                         8.000  strings (     0.000  retained)
   fast_view_assigns   688.000  memsize (     0.000  retained)
                        10.000  objects (     0.000  retained)
                         8.000  strings (     0.000  retained)

Comparison:
        view_assigns:        688 allocated
   fast_view_assigns:        688 allocated - same
```

Reply:
I have no time to investigate that for now, but as far as I can see from the code, `#[beg, len]` is through less code than `#[range]`.

`#[beg, len]`:
https://github.com/ruby/ruby/blob/7cc7627c2d4db91f1097c7f1c5f5c9a41aeaefc6/string.c#L4596-L4600

`#[range]`:
https://github.com/ruby/ruby/blob/7cc7627c2d4db91f1097c7f1c5f5c9a41aeaefc6/string.c#L4502-L4511

Reply:
You are correct. The code for `range` is more expensive. I benchmarked slicing vs using range on a string in isolation and slicing is indeed cheaper.